spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -354,6 +344,19 @@ private void applySelectWithFilter(Object[] parameters, Builder builder) {
 						: filters[0]);
 	}
 
+	private Value convertValue(Iterator it, Part part, DatastorePersistentProperty persistentProperty) {
```

Comment:
It's a bit odd to pass in an iterator here rather than the actual value.
Also the check for `IS_NULL` is completely redundant since you do it in the switch statement.